Dockerfile

```diff
@@ -0,0 +1,57 @@
+FROM --platform=linux/amd64 rust:1.79-slim-bookworm
+
+LABEL maintainer="DecideAI"
+
+ENV NVM_DIR=/root/.nvm
+ENV NVM_VERSION=v0.39.7
+ENV NODE_VERSION=21.5.0
+ENV DFX_VERSION=0.19.0
+ENV POCKET_IC_SERVER_VERSION=3.0.1
+ENV POCKET_IC_PYTHON_VERSION=2.1.0
+
+RUN apt -yq update
+RUN apt -yqq install --no-install-recommends curl ca-certificates libunwind-dev git python3 python3-pip ssh
+
+# Install Node.js using nvm
+ENV PATH="/root/.nvm/versions/node/v${NODE_VERSION}/bin:${PATH}"
+RUN curl --fail -sSf https://raw.githubusercontent.com/creationix/nvm/${NVM_VERSION}/install.sh | bash
+RUN . "${NVM_DIR}/nvm.sh" && nvm install ${NODE_VERSION}
+RUN . "${NVM_DIR}/nvm.sh" && nvm use v${NODE_VERSION}
+RUN . "${NVM_DIR}/nvm.sh" && nvm alias default v${NODE_VERSION}
+
+# Install dfx
+RUN DFXVM_INIT_YES=true sh -ci "$(curl -fsSL https://internetcomputer.org/install.sh)"
+ENV PATH="/root/.local/share/dfx/bin:$PATH"
+ENV DFX_VERSION=0.20.1
+
+# Add wasm32-unknown-unknown target
+RUN rustup target add wasm32-unknown-unknown
+
+# Install PocketIC Python
+RUN pip3 install pocket-ic==${POCKET_IC_PYTHON_VERSION} --break-system-packages
+
+# Download the PocketIC server
+RUN curl -Ls https://github.com/dfinity/pocketic/releases/download/${POCKET_IC_SERVER_VERSION}/pocket-ic-x86_64-linux.gz -o pocket-ic.gz
+RUN gzip -d pocket-ic.gz
+RUN chmod +x pocket-ic
+
+# Clean apt
+RUN apt-get autoremove && apt-get clean
+
+# Set the working directory inside the container
+WORKDIR /app
+
+COPY . .
```

Comment:
Dont copy . , otherwise we need to rebuild the Docker image for every code change.
you can map the source folder from your host machine into the Docker container using a volume.

Reply:
em. my above comment is wrong. 
we have to rebuild the wasm for any code change. 
I am just thinking if there is a way to avoid rebuilding images..

Reply:
usually for dev image, only copy package.json/requirements.txt/etc..

Reply:
This is for running tests, not for development hence the command to copy everything.

If you need to rerun the tests, you need to rebuild.

I deliberately didn't setup volumes because we don't need it